Source/lighting.cpp

```diff
@@ -921,14 +921,15 @@ int AddLight(Point position, int r)
 	}
 
 	lid = NO_LIGHT;
+	auto &lights = Lights[lid];
 
 	if (ActiveLightCount < MAXLIGHTS) {
 		lid = ActiveLights[ActiveLightCount++];

```

Comment:
If you make the reference before knowing the index (`lid`), then you will be operating on the wrong reference later on.

Also the value should only be declared where in the same scope (`{}`) so that it can't be used in the wrong place by mistake.
```suggestion

	if (ActiveLightCount < MAXLIGHTS) {
		lid = ActiveLights[ActiveLightCount++];
		Light &lights = Lights[lid];
```
(one last thing here, we decided that `auto` should preferably only by used in places where the type also appears on the right side)

Reply:
Got it!